tests/template_tests/syntax_tests/test_cycle.py

```diff
@@ -127,3 +127,39 @@ def test_cycle27(self):
     def test_cycle28(self):
         output = self.engine.render_to_string('cycle28', {'a': '<', 'b': '>'})
         self.assertEqual(output, '<&gt;')
+
+    @setup({
+        'cycle29': "{% cycle 'a' 'b' 'c' as cycler silent %}"
+                   "{% for x in values %}"
+                   "{% with doesnothing=irrelevant %}"
+                   "{% ifchanged x %}"
+                   "{% cycle cycler %}{{ cycler }}"
+                   "{% else %}"
+                   "{{ cycler }}"
+                   "{% endifchanged %}"
+                   "{% endwith %}"
+                   "{% endfor %}"})
+    def test_cycle29(self):
```

Comment:
A semantic test name or a docstring could be useful. (e.g. "A {% with %} tag shouldn't reset the {% cycle %} variable.")